src/figuresClasses.ts

```diff
@@ -1,19 +1,83 @@
-export interface Figure {
+export enum Shape {
+  Triangle = 'triangle',
+  Circle = 'circle',
+  Rectangle = 'rectangle',
+}
 
+export enum Color {
+  Red = 'red',
+  Green = 'green',
+  Blue = 'blue',
+}
+
+export interface Figure {
+  shape: Shape;
+  color: Color;
+  getArea(): number;
 }
 
 export class Triangle implements Figure {
+  shape: Shape = Shape.Triangle;
 
+  constructor(
+    public color: Color,
+    public a: number,
+    public b: number,
+    public c: number,
+  ) {
+    if (a <= 0 || b <= 0 || c <= 0 || a + b <= c || a + c <= b || b + c <= a) {
+      throw new Error('Invalid figure information');
+    }
+  }
+
+  getArea(): number {
+    const square = (this.a + this.b + this.c) / 2;
+
+    return +Math.sqrt(
+      square * (square - this.a) * (square - this.b) * (square - this.c),
+    ).toFixed(
+      2,
+    );
+  }
 }
 
 export class Circle implements Figure {
+  shape: Shape = Shape.Circle;
+
+  constructor(
+    public color: Color,
+    public radius: number,
+  ) {
+    if (radius <= 0) {
+      throw new Error('Invalid figure information');
+    }
+  }
 
+  getArea(): number {
+    return +((Math.PI * this.radius ** 2)).toFixed(2);
```

Comment:
```suggestion
    return Math.floor(Math.PI * (this.radius ** 2) * 100) / 100;
```
Try this to pass the tests